lib/parsers/stream-json-parser.ts

```
@@ -0,0 +1,229 @@
+import { Transform } from 'stream';
+import { ClaudeMessage } from '@/types/claude.types';
+
+interface ParsedMessage {
+  type: 'message' | 'error' | 'status' | 'result' | 'incomplete';
+  data: ClaudeMessage | string | any;
+  raw: string;
+}
+
+export class StreamJsonParser extends Transform {
+  private buffer: string = '';
```

Biome (1.9.4)

[error] This type annotation is trivially inferred from its initialization.

Safe fix: Remove the type annotation.


(lint/style/noInferrableTypes)